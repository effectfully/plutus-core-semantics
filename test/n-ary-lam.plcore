(program

(module Foo (imported) (exported () ()) (local () ())
  (define r1 [(lam x y y) 1 2])
  (define r2 [(lam x y z (builtin addInt 10 z)) 1 2 3])
  (define r3 [(lam x y (builtin addInt x y)) 10 20])
  (define r4 [(lam x y z (builtin addInt x (builtin addInt y z))) 10 20 5])
  (define r5 [[(lam f g [(lam x x) f]) (lam x y y) (lam x y x)] #A #B]))

)
