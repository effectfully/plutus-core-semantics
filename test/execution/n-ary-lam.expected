<T>
  <k> . </k>
  <lenv> .Map </lenv>
  <genv> ( Foo . r4 ) |-> 35 ( Foo . r5 ) |-> #B ( Foo . r2 ) |-> 13 ( Foo . r3 ) |-> 30 ( Foo . r1 ) |-> 2 </genv>
  <modules>
    <Module>
      <curr> 0 </curr>
      <name> Foo </name>
      <dataCons> .TyExpList </dataCons>
      <decl> .NameList </decl>
      <importedModules> .ModNmList </importedModules>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </Module>
  </modules>
</T>
